entitlement/src/main/java/org/killbill/billing/entitlement/api/DefaultEntitlementApi.java

```diff
@@ -332,7 +338,7 @@ public UUID doCall(final EntitlementApi entitlementApi, final DefaultEntitlement
 
                     final DefaultBaseEntitlementWithAddOnsSpecifier baseEntitlementWithAddOnsSpecifier = getFirstBaseEntitlementWithAddOnsSpecifier(updatedPluginContext.getBaseEntitlementWithAddOnsSpecifiers());
 
-                    final DateTime requestedDate = dateHelper.fromLocalDateAndReferenceTime(baseEntitlementWithAddOnsSpecifier.getBillingEffectiveDate(), updatedPluginContext.getCreatedDate(), contextWithSourceAccountRecordId);
+                    final DateTime requestedDate = baseEntitlementWithAddOnsSpecifier.getBillingEffectiveDate();
```

Comment:
@pierre Interesting, should we also add an api with `DateTime` for transfer?

Reply:
@reshmabidikar Also, I noticed that we are not always super consistent with where we get the date from: Sometimes we use the date from the `pluginContext` (-> `baseEntitlementWithAddOnsSpecifier` -> `getBillingEffectiveDate`) and sometimes we use the date built on top of the `doCall` method (as a closure). It should be the same date though, assuming we don't mess up.

(Perhaps, this was already the case prior your change).

Reply:
> @pierre Interesting, should we also add an api with `DateTime` for transfer?

LMK if this needs to be added now or later on (we had discussed this in the [original google doc](https://docs.google.com/document/d/1pE3AaQHRnu14nU2_SFnPb-RDzFxj3eTi5x4zdhHsydw/edit#heading=h.ftyog36esr8v) and decided to skip for now.

Reply:
> @reshmabidikar Also, I noticed that we are not always super consistent with where we get the date from: Sometimes we use the date from the `pluginContext` (-> `baseEntitlementWithAddOnsSpecifier` -> `getBillingEffectiveDate`) and sometimes we use the date built on top of the `doCall` method (as a closure). It should be the same date though, assuming we don't mess up.
> 
> (Perhaps, this was already the case prior your change).

Yes, I guess wherever possible, the code uses the dates passed to it (after the conversion to datetime if applicable) within the `doCall` method and in other cases uses the date from the `pluginContext`.  Do you want me to make changes so that the behavior is consistent throughout? If so, I think using the date from  `pluginContext` makes more sense. What do you think?

Reply:
I think using the dates from the pluginContext is indeed better - as we ensure that both KB core logic and plugin see the same dates. We can fix it in another PR - not super urgent, but it would help with maintainability moving forward.